libp2p/network/connection/swarm_connection.py

```diff
@@ -45,43 +51,32 @@ def __init__(self, muxed_conn: IMuxedConn, swarm: "Swarm") -> None:
             await stream.reset()
         # Force context switch for stream handlers to process the stream reset event we just emit
         # before we cancel the stream handler tasks.
-        await asyncio.sleep(0.1)
+        await trio.sleep(0.1)
 
-        for task in self._tasks:
-            task.cancel()
-            try:
-                await task
-            except asyncio.CancelledError:
-                pass
-        # Schedule `self._notify_disconnected` to make it execute after `close` is finished.
         self._notify_disconnected()
 
     async def _handle_new_streams(self) -> None:
+        self.event_started.set()
         while True:
             try:
                 stream = await self.muxed_conn.accept_stream()
+            # Asynchronously handle the accepted stream, to avoid blocking the next stream.
```

Comment:
Is this comment in the right line you intended to put?

Reply:
No, this line is wrongly indented. Thank you for pointing it out!